plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php

```diff
@@ -82,6 +83,37 @@ public function __construct() {
 		}
 	}
 
+
+	/**
+	 * Adds the product template ID to the product if it doesn't exist.
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param WC_Product       $product The product.
+	 */
+	public function possibly_add_template_id( $response, $product ) {
+		if ( ! $product ) {
+			return $response;
+		}
+		if ( ! $product->meta_exists( '_product_template_id' ) ) {
+			/**
+			 * Experimental: Allows to determine a product template id based on the product data.
+			 *
+			 * @ignore
+			 * @since 9.1.0
+			 */
+			$product_template_id = apply_filters( 'experimental_woocommerce_product_editor_product_template_for_product', '', $product );
+			if ( $product_template_id ) {
+				$response->data['meta_data'][] = new WC_Meta_Data(
+					array(
+						'key'   => '_product_template_id',
+						'value' => $product_template_id,
+					)
+				);
```

Comment:
Thanks for updating this, I did notice one small issue, now that we return it like this the meta data doesn't show up as dirty.
So saving it doesn't automatically add the `_product_template_id`, a user would actually have to switch a template manually in order to save it.
Now on the other hand this would also be fine for now as it would just re-run the logic on the back-end.
I suppose we could improve on this in the future.

Reply:
Yes, that is a downside of the change to use the `woocommerce_rest_prepare_product_object` filter. As you stated, it will fulfill the purpose, so in my opinion we can leave it like this for now.